ssoPlugin/widget/index.html

```diff
@@ -0,0 +1,278 @@
+			<button id="bif-sso__login-button">Log In</button>
+		</div>
+
+		<div id="bif-sso__content--authenticated" class="bif-sso__content">
+			<p>Successfully authenticated!</p>
+		</div>
+	</div>
+
+	<script type="text/javascript">
+		// ============================================================================================
+		// UTILITY METHODS
+		// ============================================================================================
+
+		/**
+		 * Runs the provided function when the document is fully loaded.
+		 *
+		 * @param {Function} fn The function to be run.
+		 */
+		const ready = (fn) => {
+			if (document.readyState != 'loading') fn()
+			else document.addEventListener('DOMContentLoaded', fn)
+		}
+
+		/**
+		 * Saves the provided data object to the browser's localStorage using the BuildFire library.
+		 *
+		 * @param   {Object}  data An object containing the data to save.
+		 * @returns {Promise}      Resolves with boolean true, rejects with an error message.
+		 */
+		const saveToLocalStorage = (data) => new Promise((resolve, reject) => {
+			buildfire.localStorage.setItem('@bif/sso', data, (err, result) => {
+				if (err || !result) reject('There was a problem saving to local storage.')
+				else resolve(true)
+			})
+		})
+
+		// ============================================================================================
+		// APPLICATION METHODS
+		// ============================================================================================
+
+		// Set DOM variables that we'll reference in the following methods thanks to lexical scoping.
+		const $body = document.querySelector('body')
+		const $alertMessage = document.getElementById('bif-sso__alert-message')
+		const $loginButton = document.getElementById('bif-sso__login-button')
+
+		/**
+		 * Marks the completion of initialization by changing some body classes.
+		 */
+		const initComplete = () => {
+			$body.classList.remove('uninitialized')
+			$body.classList.add('initialized')
+		}
+
+		/**
+		 * Updates the alert message placeholder with the provided message and type.
+		 *
+		 * Expected types are "error" or "success". Default is "notification" and has no special style.
+		 *
+		 * @param {String}  message      The message that should be displayed.
+		 * @param {String} messageType? The message type, from the list above.
+		 */
+		const alertMessage = (message, messageType = 'notification') => {
+			const filteredClassNames = $alertMessage.className
+				.split(' ')
+				.filter((className) => !className.startsWith('bif-sso__alert-message--'))
+			if (message) filteredClassNames.push(`bif-sso__alert-message--${messageType}`)
+			$alertMessage.className = filteredClassNames.join(' ').trim()
+			$alertMessage.innerHTML = message
+		}
+
+		/**
+		 * Validates that the required settings exist and aren't falsey.
+		 *
+		 * @param {Object} settings An object containing the settings.
+		 */
+		const areValidSettings = (settings) => {
```

Comment:
Same question here re: DRY

Reply:
Same answer: moving to a `utils.js` would I suppose be nice for getting it out of this codebase, but, to be clear we can't actually share it with other codebases until converting to a Webpacked plugin (#8).

I can still see _some_ advantages, like, at least having a common (if duplicated) set of `utils.js`. Hmm. Maybe I'll do this anyway after all, thanks for the idea.